tests/test_port_add_remove.py

```diff
@@ -0,0 +1,136 @@
+        for key in pgs:
+            if PORT in key:
+                config_db.delete_entry('BUFFER_PG', key)
+
+        num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                              num_of_ports-1,
+                              polling_config = PollingConfig(polling_interval = 1, timeout = 5.00, strict = False))
+
+        # verify that the port wasn't removed since we still have buffer cfg
+        assert len(num) == num_of_ports
+
+        # modify buffer queue entry to egress_lossless_profile instead of egress_lossy_profile
+        config_db.update_entry("BUFFER_QUEUE", "%s|0-2"%PORT, {"profile": "egress_lossless_profile"})
+
+        # remove buffer queue cfg for the port
+        pgs = config_db.get_keys('BUFFER_QUEUE')
+        for key in pgs:
+            if PORT in key:
+                config_db.delete_entry('BUFFER_QUEUE', key)
+
+        num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                              num_of_ports-1,
+                              polling_config = PollingConfig(polling_interval = 1, timeout = 5.00, strict = True))
+
+        # verify that the port was removed properly since all buffer configuration was removed also
+        assert len(num) == num_of_ports - 1
+        
+        config_db.create_entry("PORT", PORT, port_info)
+
+    @pytest.mark.parametrize("scenario", ["one_port", "all_ports"])
+    def test_add_remove_all_the_ports(self, dvs, testlog, scenario):
+        config_db = dvs.get_config_db()
+        asic_db = dvs.get_asic_db()
+     
+        # get the number of ports before removal
+        num_of_ports = len(asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT"))
+        
+        # remove buffer pg cfg for the port
+        if scenario == "all_ports":
+            ports = config_db.get_keys('PORT')
+        elif scenario == "one_port":
+            ports = [PORT]
+        else:
+            assert False
+
+        ports_info = {}
+        
+        for i in range(DELETE_CREATE_ITERATIONS):
+            # remove ports
+            for key in ports:
+                # read port info and save it
+                ports_info[key] = config_db.get_entry("PORT", key)
+                
+                # remove a port
+                self.dvs_port.remove_port(key)
+    
+            # verify remove port
+            num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                                          num_of_ports-len(ports))
+            assert len(num) == num_of_ports-len(ports)
+            
+            # add port
+            time.sleep(i%3)
+            for key in ports:
+                config_db.create_entry("PORT", key, ports_info[key])
+
+            # verify add port            
+            num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                                  num_of_ports)
+            assert len(num) == num_of_ports
+                
+            time.sleep((i%2)+1)           
```

Comment:
what is the waiting here, and why it varies?

Reply:
same as above:
different timeouts between delete/create can catch potential race conditions that can lead to system crush.